targets/TARGET_STM/TARGET_STM32WB/wb_sleep.c

```diff
@@ -58,6 +59,9 @@ void hal_deepsleep(void)
     PWR_EnterStopMode();
     PWR_ExitStopMode();
 
+    /* Force complete clock reconfiguration */
+    SetSysClock();
```

Comment:
This is a generic way indeed to restart all the clocks. But it can take time and have impacts on power consumption. What is actually missing at this point, is it only HSI48 ?
if so, isn't it enough  to call again :
```
while` (LL_HSEM_1StepLock(HSEM, CFG_HW_HSI48_SEMID));
```
of course that's an optimization only

Reply:
Idea was to ease maintenance.
If we change something in SetSysClock, it will be automatically taken into account at STOP mode exit.

Reply:
> But it can take time and have impacts on power consumption.

what we are losing here ? Any numbers?

Reply:
> > But it can take time and have impacts on power consumption.
> what we are losing here ? Any numbers?

No measure, but I don't think this takes more time than switching only HSI48 clock here.